### PR TITLE
Pull Request for Issue1974: Release changes made during SGM Deploy 2016-02-08

### PR DESCRIPTION
**Pull request by iainworkman**
- Removed dependency on CLSAMDSScalerChannelDetector from AMGenericScanActionControllerAssembler.
  - Created a virtual function generateListOfDetectorsToConfigure() in AMGenericScanActionControllerAssembler which creates the list of detectors which need to be configured. The base implementation returns a list of all detectors.
  - Created a SGMContinuousScanControllerAssembler which overrides generateListOfDetectorsToConfigure() to ensure that only one instance of the CLSAMDSScalerChannelDetectors is added.
- Added include of AMDS to AMDetector.pri
- Added QT += network to AMDS.pri
- Removed VESPERSDatabaseDuplicateEntryPatch from acquaman_suite.pro
- Remade scaler/amptek tolerance changes which had been torched when moving old code from the generic scan action controller to the SGM continuous scan controller.

**NOTE:** This will be merged into SGMUpgrade_mergedMaster, before that is merged into SGMUpgrade. SGMUpgrade will then be merged into master.

### CODE REVIEW

**Comment by iainworkman** [comment]
Hey @helfrij. Would you mind looking over these changes for me? Bear in mind these were changes made during a deploy. Some of them were to get acquaman_suite to build with all the changes made in the SGMUpgrade branch, others were more specific to getting SGMUpgrade to work with the changes made in the last week.
Thanks.

File: acquaman_suite.pro [hunk 1]
```diff
@@ -29,7 +29,7 @@ SUBDIRS += \
 	SXRMBAddOnsCoordinator.pro \
 	VESPERSAcquaman.pro \
 	VESPERSBendingMirrors.pro \
-	VESPERSDatabaseDuplicateEntryPatch.pro \
+#	VESPERSDatabaseDuplicateEntryPatch.pro \
```
Comment (helfrij) [inline acquaman_suite.pro]: Quick question--would VESPERS be negatively affected by removing this .pro file?

File: source/acquaman/AMGenericScanActionControllerAssembler.cpp [hunk 2]
```diff
@@ -30,12 +30,6 @@ along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
 #include "actions3/actions/AMDetectorTriggerAction.h"
 #include "beamline/AMDetectorTriggerSource.h"
 
-#include "beamline/CLS/CLSAMDSScalerChannelDetector.h"
-#include "beamline/SGM/SGMBeamline.h"
-#include "beamline/SGM/energy/SGMEnergyControlSet.h"
-#include "beamline/SGM/energy/SGMGratingSupport.h"
-#include "beamline/SGM/energy/SGMExitSlitSupport.h"
-
 #include <QDebug>
```
Comment (helfrij) [inline source/acquaman/AMGenericScanActionControllerAssembler.cpp]: Is this QDebug still necessary?

**Comment by helfrij** [comment]
Looks good, @iainworkman!

Time Estimate: 10 minutes.

File: acquaman_suite.pro (hunk 1 above)
Comment (dretrex) [inline acquaman_suite.pro]: Nope.

**Comment by dretrex** [comment]
Looks good to me too.  Once the debug is gone I'll merge this in.
Time Estimate: 2 minutes

**Comment by iainworkman** [comment]
Okay @dretrex this should be ready to merge in.
Once this is done I'm going to merge this into SGMUpgrade, do a double check to make sure everything necessary is in SGMUpgrade (including the deploy branch) and then once again bring master into it.

Once that's done it should be ready to merge into master. 
@acquaman/acquaman-working-group should all be aware that anything branched out from master at the time that SGMUpgrade is merged in might well have merge conflicts.

**Pull request merged**